TileStache/Mapnik.py

```diff
@@ -74,22 +74,37 @@ class ImageProvider:
         - http://trac.mapnik.org/wiki/XMLConfigReference
     """
 
-    def __init__(self, layer, mapfile, fonts=None, scale_factor=None):
-        """ Initialize Mapnik provider with layer and mapfile.
+    def __init__(self, layer, mapconfig, fonts=None, scale_factor=None):
+        """ Initialize Mapnik provider with layer and mapconfig.
 
-            XML mapfile keyword arg comes from TileStache config,
-            and is an absolute path by the time it gets here.
+            XML mapconfig keyword arg comes from TileStache config,
+            and is either an absolute path by the time it gets here or a
+            XML string.
         """
-        maphref = urljoin(layer.config.dirpath, mapfile)
-        scheme, h, path, q, p, f = urlparse(maphref)
+        self.mapnik = None
+        self.layer = layer
 
-        if scheme in ('file', ''):
-            self.mapfile = path
-        else:
-            self.mapfile = maphref
+        try:
```

Comment:
@alculquicondor the developers don't have to look at this part of code if eveything works but if we go with two separation options, we will still have to converge at some point. How about this, we merge this branch and I will make another pass on it since I would like to clean up few other thigns as well.